commerce-product-api/src/main/java/com/liferay/commerce/product/data/source/CPDataSourceResult.java

```diff
@@ -77,6 +80,16 @@ public String toString() {
 		return sb.toString();
 	}
 
+	private void readObject(ObjectInputStream stream)
```

Comment:
@alecsloan why?

Reply:
If I remember correctly it complained about it being an implementation of Serializable but not having read or write methods.

Reply:
The javadoc for `Serializable` says:

> Classes that require special handling during the serialization and deserialization process must implement special methods with these exact signatures:

    private void writeObject(java.io.ObjectOutputStream out)
         throws IOException
    private void readObject(java.io.ObjectInputStream in)
         throws IOException, ClassNotFoundException;
    private void readObjectNoData()
         throws ObjectStreamException;

So I think we should only have to include these methods if we need custom de/serialization logic, which it looks like we don't.